docs/design/features/StringDeduplication.md

```diff
@@ -0,0 +1,196 @@
+# **String Deduplication**
+
+## **Problem Space**
+
+Duplicated strings have been frequently observed in managed heaps. In some user scenarios they can take up most of the heap. Folks have been doing their own string deduplication. One of our first party customers did their own deduplication and were able to reduce the heap size by >70% but they could not do this for all workloads they have; only when they can find a feasible place to do so.
+
```

Comment:
Another reason why Roslyn does string deduplication is to speed up string equality. Useful when doing a lot of hashtable lookups with string keys.